doc: remove RHEL 7 and mention minor version support policy (#4594)
Kong/docs.konghq.com: app/gateway/2.6.x/compatibility.md

{% for system in compat_data.os %}
  <li><strong>{{ system[0] | split: "-" | join: " " | capitalize }}:</strong> {{ system[1] }}</li>
{% endfor %}

> For RHEL and CentOS, only the minor version that the vendor supports is supported by Kong.

Reviewer: ```suggestion
Kong only supports the latest vendor-supported minor version for RHEL and CentOS.
```
